### PR TITLE
hsapp: remove hsinput; use giu implementation of imput handler

### PR DESCRIPTION
**Pull request by gucio321**
Hi there,
handling input events deffinitely isn't a part of HellSpawner. since 1dfaa030a55f5ec37408bc26544e6666180fb9e0 `giu` supports that so we can use `built-in` input manager.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -47,7 +47,6 @@ linters:
     - goconst
     - gocritic
     - gocyclo
-    - godox
     - gofmt
     - goimports
     - gomnd
```
Comment (gravestench) [inline .golangci.yml]: why are you changing the golint config in this PR?

File: hswindow/hseditor/hsdt1editor/dt1_editor.go [hunk 2]
```diff
@@ -113,26 +112,29 @@ func (e *DT1Editor) UpdateMainMenuLayout(l *g.Layout) {
...
+
+			// left arrow goes to the previous tile group
+			giu.WindowShortcut{
+				Callback: func() {
+					e.dt1Viewer.SetTileGroup(e.dt1Viewer.TileGroup() - 1)
+				},
+				Key:      g.KeyLeft,
+				Modifier: g.ModNone,
+			},
+		*/
```
Comment (gravestench) [inline hswindow/hseditor/hsdt1editor/dt1_editor.go:137]: should we delete this?

**Review by gravestench — CHANGES_REQUESTED** [review]
please revert `.golangci.yml`

File: .golangci.yml (hunk 1 above)
Comment (gucio321) [reply .golangci.yml]: 2a1fd3c and 73e7c1d  reverted this

File: hswindow/hseditor/hsdt1editor/dt1_editor.go (hunk 2 above)
Comment (gucio321) [reply hswindow/hseditor/hsdt1editor/dt1_editor.go:137]: not really sure. But this code was here before so it isn't a part of this PR. I opened an issue: https://github.com/OpenDiablo2/HellSpawner/issues/329.

Comment (gucio321) [reply hswindow/hseditor/hsdt1editor/dt1_editor.go:136]: not really sure... we may want to use this code. I'll open an issue for it

Comment (gravestench) [reply hswindow/hseditor/hsdt1editor/dt1_editor.go:137]: okay, as long as we have an issue for it this is fine.

**Review by gravestench — APPROVED** [review]

File: hswindow/hsdialog/dialog.go [hunk 3]
```diff
@@ -32,9 +30,14 @@ func (d *Dialog) Show() {
...
-	// noop
+// RegisterKeyboardShortcuts registers a shortcuts for popup dialog
+func (d *Dialog) RegisterKeyboardShortcuts(_ ...giu.WindowShortcut) {
+	// https://github.com/OpenDiablo2/HellSpawner/issues/327
+}
+
+// KeyboardShortcuts returns a list of shortcuts
+func (d *Dialog) KeyboardShortcuts() []giu.WindowShortcut {
+	return []giu.WindowShortcut{}
 }
```
Comment (gravestench) [inline hswindow/hsdialog/dialog.go:41]: If we wanted the dialog modal to close when pressing escape, this is where we would define that, right?

File: hswindow/hseditor/hsdc6editor/dc6_editor.go [hunk 4]
```diff
@@ -110,14 +109,6 @@ func (e *DC6Editor) UpdateMainMenuLayout(l *g.Layout) {
 	*l = append(*l, m)
 }
 
-// RegisterKeyboardShortcuts adds a local shortcuts for this editor
-func (e *DC6Editor) RegisterKeyboardShortcuts(inputManager *hsinput.InputManager) {
-	// Ctrl+Shift+S saves file
-	inputManager.RegisterShortcut(func() {
-		e.Save()
-	}, g.KeyS, g.ModShift+g.ModControl, false)
-}
```
Comment (gravestench) [inline hswindow/hseditor/hsdc6editor/dc6_editor.go:119]: is this `ctrl`+`shift`+`s` hotkey being registered elsewhere, now?

**Review by gravestench — APPROVED** [review]

File: hswindow/hsdialog/dialog.go (hunk 3 above)
Comment (gucio321) [reply hswindow/hsdialog/dialog.go:41]: no, if we'd like to add shortcut to some specified dialog we need to handle it here. Closing dialogs by escape is currently done here: https://github.com/OpenDiablo2/HellSpawner/blob/2a0e033f27d6296bda1b1714b8ee5cef05d5e672/hsapp/setup.go#L260

File: hswindow/hseditor/hsdc6editor/dc6_editor.go (hunk 4 above)
Comment (gucio321) [reply hswindow/hseditor/hsdc6editor/dc6_editor.go:119]: it isn't, however, I'm not really sure if this shortcut is necessary. saving editor data takes place on ctrl+s and this shortcut is registered in hsapp.

**Pull request merged**